bgpd/bgpd.c

```diff
@@ -3070,6 +3070,18 @@ struct bgp *bgp_lookup_by_vrf_id(vrf_id_t vrf_id)
 	return (vrf->info) ? (struct bgp *)vrf->info : NULL;
 }
 
+/* Returns the BGP instance where EVPN is enabled, if any */
+struct bgp *bgp_get_evpn(void)
+{
+	struct bgp *bgp;
+	struct listnode *node, *nnode;
+
+	for (ALL_LIST_ELEMENTS(bm->bgp, node, nnode, bgp))
+		if (bgp->advertise_all_vni)
```

Comment:
what if multiple vrfs have enabled this option?

Reply:
This is prevented by 44fbaa344b74262ba7c47b0dcb3cec02e24a39d5 which restricts the option to a single VRF.

Reply:
Ooops I'm sorry, I meant to link 53aac248652deb339cc0df6e366f5ce419dbad3d, specifically this part : https://github.com/FRRouting/frr/pull/3923/commits/53aac248652deb339cc0df6e366f5ce419dbad3d#diff-c3e97f9bf7983d6fc51ceba93358b015R2963